sycl/include/sycl/ext/oneapi/experimental/annotated_ptr/annotated_ptr.hpp

```diff
@@ -393,14 +396,17 @@ __SYCL_TYPE(annotated_ptr) annotated_ptr<T, detail::properties_t<Props...>> {
   reference operator*() const noexcept { return reference(m_Ptr); }
 
   reference operator[](std::ptrdiff_t idx) const noexcept {
+    operatorAvailablityCheck(op::subscript);
     return reference(m_Ptr + idx);
   }
 
   annotated_ptr operator+(size_t offset) const noexcept {
```

Comment:
Binary operators should normally be symmetric. And they are for the similar case of [random_access_iterator](https://en.cppreference.com/w/cpp/iterator/random_access_iterator). Needs to be changed to two hidden friends. Needs to be fixed in the spec too. Can be done in a separate PR.

Reply:
Will refactor this part (maybe in a separate PR) when all the comments in this PR are resolved